File: app/src/main/java/com/qianxinde/libtest/utils/ImageViewAdapter.java
package com.qianxinde.libtest.utils;

import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;
import android.widget.ImageView;

import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;
import com.qianxinde.libtest.R;

/**
 * @author yangbw
 */
public class ImageViewAdapter {

    @BindingAdapter("android:src")
    public static void setSrc(ImageView view, Bitmap bitmap) {
        view.setImageBitmap(bitmap);
    }
 
    @BindingAdapter("android:src")
    public static void setSrc(ImageView view, int resId) {
        view.setImageResource(resId);
    }

    @BindingAdapter("imageUrl")
    public static void setSrc(ImageView imageView, String url) {
        Glide.with(imageView.getContext()).load(url)
                .placeholder(R.mipmap.ic_launcher)
                .into(imageView);
    }

    @BindingAdapter({"app:imageUrl", "app:placeHolder", "app:error"})
    public static void loadImage(ImageView imageView, String url, Drawable holderDrawable, Drawable errorDrawable) {
        Glide.with(imageView.getContext())
                .load(url)
                .placeholder(holderDrawable)
                .error(errorDrawable)
                .into(imageView);
    }

}

File: app/src/main/java/com/yangbw/libtest/ui/adapter/ImageViewAdapter.kt
package com.yangbw.libtest.ui.adapter

import android.graphics.Bitmap
import android.graphics.drawable.Drawable
import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide
import com.library.common.utils.GlideUtils

/**
 * @author yangbw
 */
object ImageViewAdapter {

    @JvmStatic
    @BindingAdapter("android:src")
    fun setSrc(view: ImageView, bitmap: Bitmap?) {
        view.setImageBitmap(bitmap)
    }

    @JvmStatic
    @BindingAdapter("android:src")
    fun setSrc(view: ImageView, resId: Int) {
        view.setImageResource(resId)
    }

    @JvmStatic
    @BindingAdapter("loadImage")
    fun loadImage(imageView: ImageView, url: String?) {
        url?.let {
            GlideUtils.loadImage(imageView, it)
        }
    }

    @JvmStatic
    @BindingAdapter("loadCircleImage")
    fun loadCircleImage(imageView: ImageView, url: String?) {
        url?.let {
            GlideUtils.loadCircleImage(imageView, it)
        }
    }

    @JvmStatic
    @BindingAdapter("loadRoundImage")
    fun loadRoundImage(imageView: ImageView, url: String?) {
        url?.let {
            GlideUtils.loadRoundImage(imageView, it)
        }

    }

    @JvmStatic
    @BindingAdapter("imageUrl", "placeHolder", "error")
    fun loadImage(
        imageView: ImageView,
        url: String?,
        holderDrawable: Drawable?,
        errorDrawable: Drawable?
    ) {
        Glide.with(imageView.context)
            .load(url)
            .placeholder(holderDrawable)
            .error(errorDrawable)
            .into(imageView)
    }
}